### PR TITLE
chore: Updated q, superagent, and when tests to node:test

### LINKED ISSUE #2734: Migrate `q`, `when` and `superagent` versioned tests to `node:test`

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #2734.

### CODE REVIEW

File: test/versioned/when/segments.test.js [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+        process.nextTick(transaction.end.bind(transaction))
+      })
+    })
+
+    await plan.completed
+  })
+
+  // Note: this test was originally a `promise_segments: false` test listed
+  // under the segments enabled suite.
+  await t.test('then handler that returns a new promise', async (t) => {
```
Comment (jsumners-nr) [inline test/versioned/when/segments.test.js:86]: This test was originally https://github.com/newrelic/node-newrelic/blob/dffdac7f450a23c6e23a173c91bda583d40e573d/test/versioned/when/legacy-promise-segments.js#L67-L78

I don't think it should have been under the "promise_segments: true" group to begin with.

File: test/versioned/when/segments.test.js [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+          process.nextTick(transaction.end.bind(transaction))
+        })
+    })
+
+    await plan.completed
+  })
+
+  // Note: this test was originally a `promise_segments: false` test listed
+  // under the segments enabled suite.
+  await t.test('catch handler with error from subsequent promise', async (t) => {
```
Comment (jsumners-nr) [inline test/versioned/when/segments.test.js:189]: This one was originally https://github.com/newrelic/node-newrelic/blob/dffdac7f450a23c6e23a173c91bda583d40e573d/test/versioned/when/legacy-promise-segments.js#L144-L156

It also should not have been under `promise_segments: true`.

**Review by bizob2828 — APPROVED** [review]
LGTM, thanks for cleaning up the when tests

**Comment by jsumners-nr** [comment]
> LGTM, thanks for cleaning up the when tests

You're welcome. Really, I just couldn't even do the conversion without doing it 😭

**Pull request merged**